src/zarr/core/metadata/v3.py

```diff
@@ -251,10 +250,10 @@ def __init__(
         self,
         *,
         shape: Iterable[int],
-        data_type: npt.DTypeLike | DataType,
+        data_type: DTypeWrapper,
```

Comment:
Feels a little weird to have the word `Wrapper` in here. We're still specifying the dtype, it's just zarr's representation of the dtype.

Reply:
the problem I had with naming is that we will eventually unwrap the zarr dtype, revealing the underlying native (e.g., numpy) dtype. Both the wrapper object and the native dtype object will be in scope, so we need to give them different names. So I opted to refer to the wrapper as a wrapper as much as possible, to avoid confusing it with with native dtype. But I'd be up for alternative naming schemes that can avoid confusion.